cmd/oc-mirror/main.go

```diff
@@ -15,3 +39,47 @@ func checkErr(err error) {
 		kcmdutil.CheckErr(err)
 	}
 }
+
+func isV2() bool {
+	return len(os.Args) > 0 && slices.Contains(os.Args[:], "--v2")
+}
+
+func runOcMirrorV2Cmd(args []string) error {
```

Comment:
@r4f4 - This is clever,  it also ensures as you mentioned separating v1 and v2 go.mods